pkg/frontend/v1/frontend_test.go

```diff
@@ -227,6 +228,65 @@ func TestFrontendMetricsCleanup(t *testing.T) {
 	testFrontend(t, defaultFrontendConfig(), handler, test, nil, reg)
 }
 
+func TestFrontendStats(t *testing.T) {
```

Comment:
there's very little to test in pkg/frontend/v1/frontend.go. Isn't this test testing more or less the same thing as `TestHandler_ServeHTTP`? I added some tests in https://github.com/grafana/mimir/pull/6473 which make it possible to assert that the actual logged value - see `TestHandler_LogsFormattedQueryDetails`. Do you think that can be reused?

Reply:
This test is slightly different from `TestHandler_ServeHTTP` in that handler modifies value in the stats, and then we check if it was reported correctly. I admit that it would probably be better if it was moved to `handler_test.go` and tested other reported values from stats too. If you agree, I would do that.

`TestHandler_LogsFormattedQueryDetails` indeed seems to be similar, although it's testing `QueryDetails`, not stats.

Reply:
a QueryDetails also contains QueryStats for this purpose. I tend to prefer more complicated and elaborate unit tests instead of individual ones. I'll leave this up to

Reply:
I'll merge this as-is for now, and once https://github.com/grafana/mimir/pull/6473 is in, I will try to take a closer look at reusing `TestHandler_LogsFormattedQueryDetails`.